src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -3838,8 +3839,8 @@ class CommonVisitor : public AST::BaseVisitor<Struct> {
                 );
                 throw SemanticAbort();
             }
-            ASR::expr_t *index = nullptr;
             if (ASR::is_a<ASR::Dict_t>(*type)) {
+                this->visit_expr(*m_slice);
```

Comment:
It looks like we already have `this->visit_expr(*m_slice);` on line 3829. So, this seems redundant.